src/ray/raylet/node_manager.cc

```diff
@@ -931,65 +853,75 @@ void NodeManager::ProcessGetTaskMessage(
   const ClientID &local_client_id = gcs_client_->client_table().GetLocalClientId();
   cluster_resource_map_[local_client_id].SetLoadResources(
       local_queues_.GetResourceLoad());
+
+  // Reply would be sent in when already assigned task to a worker.
+  get_task_requests_.emplace(worker_id, std::make_pair(reply, send_reply_callback));
+
   // Call task dispatch to assign work to the new worker.
   DispatchTasks(local_queues_.GetReadyTasksWithResources());
 }
 
-void NodeManager::ProcessDisconnectClientMessage(
-    const std::shared_ptr<LocalClientConnection> &client, bool intentional_disconnect) {
-  std::shared_ptr<Worker> worker = worker_pool_.GetRegisteredWorker(client);
+/// Handle a `DisconnectClient` request.
+void NodeManager::HandleDisconnectClientRequest(
+    const rpc::DisconnectClientRequest &request, rpc::DisconnectClientReply *reply,
+    rpc::SendReplyCallback send_reply_callback) {
+  const WorkerID worker_id = WorkerID::FromBinary(request.worker_id());
+  bool intentional_disconnect = request.intentional();
```

Comment:
not sure if I get this correctly, but if this is a `disconnect` request from a worker, then it should be `intentional`?

Reply:
yes, it's not used here and i add it just because the previous implementation.

Reply:
let's just remove it.